data/mods/Xedra_Evolved/eocs/abyssal_hunger_eocs.json

```diff
@@ -0,0 +1,129 @@
+      {
+        "u_spawn_monster": "mon_bashing_globule",
+        "real_count": { "math": [ "rng( 2, 4 ) + abyssal_hunger_level" ] },
+        "min_radius": 7,
+        "max_radius": 30
+      },
+      {
+        "u_spawn_monster": "mon_grasping_globule",
+        "real_count": { "math": [ "rng( 2, 4 ) + abyssal_hunger_level" ] },
+        "min_radius": 7,
+        "max_radius": 30
+      },
+      {
+        "u_spawn_monster": "mon_spewing_globule",
+        "real_count": { "math": [ "abyssal_hunger_level - rng( 1, 2 )" ] },
+        "min_radius": 7,
+        "max_radius": 30
+      },
+      { "u_message": "You move through the monster's innards and reach another empty space." }
+    ],
+    "false_effect": [
+      { "math": [ "abyssal_hunger_level", "++" ] },
+      { "mapgen_update": "abyssal_hunger_cleanup", "target_var": { "global_val": "dungeon_loc" } },
+      { "u_teleport": { "global_val": "dungeon_start" }, "fail_message": "Something is very wrong!", "force": true },
+      { "mapgen_update": "abyssal_hunger_brain", "target_var": { "global_val": "dungeon_loc" } },
+      {
+        "u_spawn_monster": "mon_bashing_globule",
+        "real_count": { "math": [ "rng( 1, 2 ) + abyssal_hunger_level" ] },
+        "min_radius": 7,
+        "max_radius": 30
+      },
+      {
+        "u_spawn_monster": "mon_grasping_globule",
+        "real_count": { "math": [ "rng( 1, 2 ) + abyssal_hunger_level" ] },
+        "min_radius": 7,
+        "max_radius": 30
+      },
+      {
+        "u_spawn_monster": "mon_spewing_globule",
+        "real_count": { "math": [ "abyssal_hunger_level - rng( 3, 4 )" ] },
+        "min_radius": 7,
+        "max_radius": 30
+      },
+      { "u_message": "You reach the Abyssal Hunger's brain." }
+    ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_ABYSSAL_HUNGER_GRANT_SHARD",
+    "condition": { "math": [ "abyssal_hunger_level", "!=", "0" ] },
+    "effect": [ { "run_eocs": "EOC_ABYSSAL_HUNGER_EXIT" }, { "u_spawn_item": "abyssal_hunger_shard" } ]
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "EOC_ABYSSAL_HUNGER_EXIT",
+    "condition": { "math": [ "abyssal_hunger_level", "!=", "0" ] },
+    "effect": [
+      { "mapgen_update": "abyssal_hunger_cleanup", "target_var": { "global_val": "dungeon_loc" } },
+      {
+        "u_teleport": { "global_val": "original_loc" },
+        "success_message": "You return to where you performed the ritual.",
+        "fail_message": "Something is very wrong!",
+        "force": true
+      },
+      { "alter_timed_events": "abyssal_hunger" },
+      { "math": [ "abyssal_hunger_level", "=", "0" ] }
+    ]
+  },
+  {
+    "id": "EOC_DEATH_ABYSSAL_HUNGER",
+    "type": "effect_on_condition",
+    "eoc_type": "AVATAR_DEATH",
+    "condition": { "math": [ "abyssal_hunger_level", "!=", "0" ] },
+    "effect": [
+      {
+        "u_message": "It is unclear if you were slain by an immune system or digested but suffice to say, you are no more.",
```

Comment:
Note to self, apply a memorial_message or log here as able.  Might require setting an effect that grants the memorial log.